core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -621,6 +621,49 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
 object InputPSBTMap extends PSBTMapFactory[InputPSBTRecord, InputPSBTMap] {
   import org.bitcoins.core.psbt.InputPSBTRecord._
 
+  def fromWitness(witness: ScriptWitness): InputPSBTMap = {
+    witness match {
+      case p2wpkh: P2WPKHWitnessV0 =>
+        val sig = PartialSignature(p2wpkh.pubKey, p2wpkh.signature)
+        InputPSBTMap(Vector(sig))
+      case p2wsh: P2WSHWitnessV0 =>
+        val sigs = p2wsh.pubKeys.zip(p2wsh.signatures).map {
+          case (pubkey, sig) =>
+            PartialSignature(pubkey, sig)
+        }
+        InputPSBTMap(sigs.toVector)
+      case EmptyScriptWitness =>
+        InputPSBTMap.empty
+    }
+  }
+
+  @tailrec
+  def fromScriptSig(scriptSignature: ScriptSignature): InputPSBTMap = {
+    scriptSignature match {
+      case p2pkh: P2PKHScriptSignature =>
+        val sig = PartialSignature(p2pkh.publicKey, p2pkh.signature)
+        InputPSBTMap(Vector(sig))
+      case p2sh: P2SHScriptSignature =>
+        val sigs = p2sh.publicKeys.zip(p2sh.signatures).map {
+          case (key, sig) =>
+            PartialSignature(key, sig)
+        }
+        InputPSBTMap(sigs.toVector)
+      case cond: ConditionalScriptSignature =>
+        fromScriptSig(cond.nestedScriptSig)
+      case _: CLTVScriptSignature | _: CSVScriptSignature |
```

Comment:
no partial signatures?

Reply:
I don't think we have a way to get the pubkeys for these types

Reply:
I see :/ so we are able to recover public keys from signatures in ECDSA (using `CryptoUtil.recoverPublicKey`) but it will return two possible candidates, only one of which will be the correct key. Any way we can use this to not lose information here?

Also, for `LockTimeScriptSignature`s, you should be able to call `fromScriptSig` on the nested script sig right?

Reply:
To use `CryptoUtil.recoverPublicKey` we need the message they are signing, for that we need the previous output :/